x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -123,14 +123,26 @@ protected void closeInternal() {
     @Nullable
     private volatile FileChannelReference channelRef;
 
-    public CacheFile(String description, long length, Path file, Runnable onNeedFSync) {
-        this.tracker = new SparseFileTracker(file.toString(), length);
-        this.description = Objects.requireNonNull(description);
+    public CacheFile(CacheKey cacheKey, long length, Path file, Runnable onNeedFSync) {
+        this(cacheKey, new SparseFileTracker(file.toString(), length), file, onNeedFSync);
+    }
+
+    public CacheFile(CacheKey cacheKey, long length, Path file, SortedSet<Tuple<Long, Long>> ranges, Runnable onNeedFSync) {
+        this(cacheKey, new SparseFileTracker(file.toString(), length, ranges), file, onNeedFSync);
+    }
+
+    private CacheFile(CacheKey cacheKey, SparseFileTracker tracker, Path file, Runnable onNeedFSync) {
+        this.cacheKey = Objects.requireNonNull(cacheKey);
+        this.tracker = Objects.requireNonNull(tracker);
         this.file = Objects.requireNonNull(file);
         this.needsFsyncRunnable = Objects.requireNonNull(onNeedFSync);
         assert invariant();
     }
 
+    public CacheKey getCacheKey() {
```

Comment:
I expect this to also be useful later when indexing information about the cache file (snapshot, index id etc)